### PR TITLE
Docs, and build-env-dive

### PR DESCRIPTION
**Pull request by psakievich**
I'm adding some documentation and another convenience script for diving into the build environment, given a spec.  @tasmith4 I'm especially interested in your input on the latter since I think this has been a requested feature from the stk team.

### CODE REVIEW

**Review by jrood-nrel — APPROVED** [review]

File: start.sh [hunk 1]
```diff
@@ -185,3 +185,18 @@ function remove-spack-prompt() {
     unset SPACK_OLD_PS1
   fi
 }
+
+# function for diving into the build environment from a spack build
+function sm-dive() {
```
Comment (tasmith4) [inline start.sh]: So, in Sierra World, the acronym "SM" means something else ... could we consider another name for this?

**Review by tasmith4 — CHANGES_REQUESTED** [review]
other than the name, I think this looks great!

File: start.sh (hunk 1 above)
Comment (psakievich) [reply start.sh]: any suggestions?  build-env-dive?

Comment (tasmith4) [reply start.sh]: @psakievich I actually like build-env-dive a lot

Comment (psakievich) [reply start.sh]: okay `build-env-dev` it shall be

File: start.sh [hunk 2]
```diff
@@ -185,3 +184,17 @@ function remove-spack-prompt() {
...
   fi
 }
+
+# function for diving into the build environment from a spack build
+function build-env-dive() {
+  if [[ "$1" == "-h" || "$1" == "--help" ]]; then
+    echo "This command will move to the build directory for the spec provided, and then dive into the build environment in a new subshell."
+  fi
+  if [[ -z ${SPACK_ENV} ]]; then
+    echo "You must have an active environment to use sm-dive."
```
Comment (tasmith4) [inline start.sh]: also needs a name change here

**Review by tasmith4 — CHANGES_REQUESTED** [review]

**Review by tasmith4 — APPROVED** [review]

**Pull request merged**